projects/lib/src/agent/desktop-agent.ts

```diff
@@ -851,10 +851,18 @@ export class DesktopAgentImpl extends DesktopAgentProxy implements DesktopAgent
         }
         this.proxyLog('OpenRequest application resolved', LogLevel.DEBUG, { application, source });
 
-        const validStrategies: IOpenApplicationStrategy[] = await Promise.all(
-            this.openStrategies.filter(async strategy => await this.canStrategyOpenApp(application, strategy)),
+        const strategyCanOpenResults = await Promise.all(
```

Comment:
Does this code require and fault tollerence.  What happens if one of these fails?